app/components/ContactUsSection/ContactUsSection.tsx

```diff
@@ -1,43 +1,57 @@
 "use client";
 
+import { FunctionComponent, useState } from "react";
 import { useForm } from "react-hook-form";
-import { Button, H2, Input, Section, Span } from "components";
+import { ButtonWithLoading, H2, Input, Section, Span } from "components";
 import { IContactUsForm } from "./interfaces/IContactUsForm";
 import { zodResolver } from "@hookform/resolvers/zod";
+import { cn } from "utils/cn";
 import { ContactUsSchema } from "../utils/validation/ContactUsSchema";
 import sendEmailService from "services/SendEmailService";
+import { toast } from "react-toastify";
 
-import styles from "./contactUsSection.module.scss";
+type TProps = {
+  isMainPage?: boolean;
+};
+
+export const ContactUsSection: FunctionComponent<TProps> = ({ isMainPage }) => {
+  const buttonBgColor = isMainPage ? "bg-accent" : "bg-black";
+  const containerBgColor = isMainPage ? "bg-black" : "bg-accent";
+  const containerClassName = cn(
+    "py-[60px] px-4 flex flex-col items-center gap-[60px] rounded-2xl",
+    containerBgColor,
+  );
+  const [isLoading, setIsLoading] = useState(false);
 
-export const ContactUsSection = () => {
   const { register, handleSubmit: validateBeforeSubmit } =
     useForm<IContactUsForm>({
       resolver: zodResolver(ContactUsSchema),
     });
 
   const handleSubmit = async (data: IContactUsForm) => {
     try {
+      setIsLoading(true);
       const sendEmailResult = await sendEmailService.sendEmail(data);
 
       if (sendEmailResult.status === "success") {
-        console.log("Email sent successfully!");
+        toast("Email sent successfully!", { type: "success" });
       } else {
-        throw new Error("Email sending failed");
+        toast("Email sending failed", { type: "error" });
       }
     } catch (error) {
       console.error(error);
-
-      return;
+    } finally {
+      setIsLoading(false);
     }
   };
```

Comment:
Key points of refactoring here:
1. Remove `const [isLoading, setIsLoading] = useState(false);`
2. Add the `useTransition` function

```typescript
const [isPending, startTransition] = useTransition();

const handleSubmit = async (data: IContactUsForm) => {
  startTransition(async () => {
    try {
      const sendEmailResult = await sendEmailService.sendEmail(data);

      if (sendEmailResult.status === "success") {
        toast("Email sent successfully!", { type: "success" });
      } else {
        toast("Email sending failed", { type: "error" });
      }
    } catch (error) {
      console.error(error);
    }
  });
};
```